PracticeWPF/PracticeWPF/MainWindow.xaml

```diff
@@ -0,0 +1,80 @@
+﻿<Window x:Class="PracticeWPF.MainWindow"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+        xmlns:local="clr-namespace:PracticeWPF"
+        xmlns:view="clr-namespace:PracticeWPF.View"
+        xmlns:model="clr-namespace:PracticeWPF.Model"
+        xmlns:viewModel="clr-namespace:PracticeWPF.ViewModel"
+        mc:Ignorable="d"
+        Title="MainWindow" Height="450" Width="800">
+    <Window.DataContext>
+        <viewModel:MainWindowViewModel/>
+    </Window.DataContext>
+    <Window.Resources>
+        <DataTemplate x:Key="ProductTemplate" 
```

Comment:
No need to specify and key and DataType.
Key allows us to use StaticResource explicitly. Otherwise DataType allows WPF connect DataTemplate implicitly.

Reply:
It would be great if you moved controls in DataTemplate to a separate View.